modules/core/shared/src/main/scala/gem/GuideEnvironment.scala

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+
+import cats._
+import cats.implicits._
+
+sealed trait GuideEnvironment extends Product with Serializable
+
+object GuideEnvironment {
+
+  final case class Flamingos2(options: GuideOptions[GuideGroup.Flamingos2]) extends GuideEnvironment
+  final case class GmosNorth(options: GuideOptions[GuideGroup.GmosNorth ])  extends GuideEnvironment
+  final case class GmosSouth(options: GuideOptions[GuideGroup.GmosSouth ])  extends GuideEnvironment
+
+  // placeholder until the other cases are defined
+  case object NoGuiding extends GuideEnvironment
+
+  object Flamingos2 {
+    val empty: Flamingos2 =
+      Flamingos2(GuideOptions.empty[GuideGroup.Flamingos2])
+
+    def typed(o: Option[GuideEnvironment]): Flamingos2 =
+      o match {
+        case Some(e @ Flamingos2(_)) => e
+        case _                       => empty
+      }
```

Comment:
So this is problematic because it eats an error case where we need a `Flamingos2` environment but are given a `Some(GmosNorth(opts))` for example.  I could make this `unsafe` I suppose or it could return an `Option[Flamingos2]` or whatever but that pushes the problem into `TargetEnvironment`.

Reply:
Yeah this seems weird. Is it just used for reading from the database? Is it always an error to pass the wrong thing? I think I might prefer `def narrow(g: GuideEnvironment): Option[Flamingos2]` or `def narrow[F[_]: ApplicativeError[_[_], Throwable]: F[Flamingos2]` here, or `unsafe...` over in the db code somewhere.

Reply:
I'd like to move it out into the db code but I need to do something for the `TargetEnvironment` constructors like `forAsterism`:

```scala
object TargetEnvironment {

  def fromAsterism(
    a: Asterism,
    g: Option[GuideEnvironment],
    u: SortedSet[UserTarget]
  ): TargetEnvironment =
...
```
Here the `Asterism` and `GuideEnvironment` instruments may not match and it isn't clear what to do in that case.  This is where I began to suspect that I was "doing it wrong". (Okay, actually I always start out with that suspicion.)